_projects/automate_vulture_whitelist.md

```diff
@@ -0,0 +1,78 @@
+---
+name: "Automate Whitelist creation for vulture"
+collaborating_projects:
+  - coala
+  - vulture
+desc: "Automatically create whitelists and detect false positives for vulture."
+developers_involved:
+  - RJ722
+mentors:
+  - AbdealiJK
+  - jendrikseipp
+  - The-Compiler
+difficulty: medium
+status: in_progress
+initiatives:
+  - GSoC
+issues: []
+markdown: automate_vulture_whitelist.md
+requirements:
+  - "Familiarity with vulture and how whitelists work."
+tags:
+  - Vulture
+  - AST
+  - coverage
+---
+
+The aim of this project is to automatically create whitelists (to be utilized
+by [vulture](https://github.com/jendrikseipp/vulture)) for projects using sip
+files to communicate between C++ level API and Python programs, most
+significant of them being PyQt projects.
+
+#### IDEAS
+
+* Create a script which takes sip files as input and spits a whitelist with all
+  the virtual functions defined in the sip files. (Add Documentation and tests)
+* Use that script for generating a whitelist for popular projects using PyQt
+  (like [qutebrowser](https://github.com/qutebrowser/qutebrowser)).
+* Allow passing coverage.xml to vulture via `--make-whitelist` and let it find
+  the list of false-positive functions. (Add Documentation and tests as well)
+* Write a whitelist generator for shiboken, the binding
+  generator used for PySide2
+
+#### MILESTONES
+
+##### COMMUNITY BONDING
+* Discussion on whether to implement a new Tracer or to directly consume `xml`
+  output from coverage.py
+* Get familiar with coverage.xml
+* Take up small tasks from
+  [TODO](https://github.com/jendrikseipp/vulture/tree/master/TODO.rst)
+* [Follow up milestone](https://gitlab.com/coala/GSoC/GSoC-2018/milestones/28)
+
+##### CODING PHASE 1
+* The script for creating a whitelist from sip files should be implemented.
+* It should reside in it's own repository, with CI enabled for testing on both
+  the GNU/Linux and Windows platforms.
+* It should be gracefully documented.
+* Use this script to create a whitelist for
+  [qutebrowser](https://github.com/qutebrowser/qutebrowser).
+* Look at methods to increase coverage for vulture from it's already excellent
```

Comment:
"Look at methods" is not a deliverable.
Try:

Raise the coverage of vulture to 100% with limited use of `#pragma: no cover` where agreed upon by the vulture maintainers.

Reply:
I don't think that @jendrikseipp agrees. He feels like code readability decreases by littering the code base with comments.

cc @jendrikseipp

Reply:
That's true. It's no problem if code coverage is less than 100% in my opinion. But trying to raise it is a good idea.

Reply:
If we want a deliverable, we may say that: "Maximise code coverage for vulture without littering it with comments." Do you agree @jayvdb?

Reply:
If using `#pragma: no cover` to reach 100%, it shouldnt be a style concern as they should be used exceedingly sparingly, in the very difficult to reach parts of the codebase, and having a specific machine readable comment in those regions helps people reading the code know that those lines are *special*, as they are not covered, and are difficult to reach in the test suite -- i.e. dragons be there; dont hack this or merge related changes before you've had a strong coffee.

coalib (coala core) only has 21 of them, and we have PRs to remove more of them.  coverage also allows you to use a different comment text, if you dont like `#pragma`.

Anyway .. it isnt a coala project, so the decision here needs to be made by the maintainer .   "Maximise code coverage for vulture" is much better than "Look at .." :+1: